python/pycrdt/text.py

```diff
@@ -72,22 +72,22 @@ def _check_slice(self, key: slice) -> tuple[int, int]:
         else:
             start = key.start
         if key.stop is None:
-            stop = len(self) - start
+            length = len(self) - start
         elif key.stop < 0:
             raise RuntimeError("Negative stop not supported")
         else:
-            stop = key.stop - start
-        return start, stop
+            length = key.stop - start
+        return start, length
 
     def __delitem__(self, key: int | slice) -> None:
         with self.doc.transaction() as txn:
             self._forbid_read_transaction(txn)
             if isinstance(key, int):
                 self.integrated.remove_range(txn._txn, key, 1)
             elif isinstance(key, slice):
-                start, stop = self._check_slice(key)
-                if start != stop:
-                    self.integrated.remove_range(txn._txn, start, stop)
+                start, length = self._check_slice(key)
+                if length > 0:
+                    self.integrated.remove_range(txn._txn, start, length)
```

Comment:
I'd rather have `_check_slice` still return `start, stop` and compute the `length` here, as we might support `step` in the future.